pkg/codegen/pcl/binder_schema.go

```diff
@@ -566,7 +567,15 @@ func GenEnum(
 		contract.Assertf(ok,
 			"We have determined %s is a safe enum, which we define as "+
 				"being able to calculate a member for", t)
-		safeEnum(member)
+		if member != nil {
+			safeEnum(member)
+		} else {
+			unsafeEnum(from)
+			*diags = append(*diags, &hcl.Diagnostic{
+				Severity: hcl.DiagError,
+				Summary:  fmt.Sprintf("\"%v\" is an invalid enum value.", known.AsString()),
```

Comment:
1. Instead of saying an "invalid enum value", can we mention the type. `"block" is an invalid enum value.` tells me that `"block"` is the problem. I think that `"block" is not a valid instance of the enum "azure-native:storage:BlobType"` is much clearer. Especially when combined with (3).

2. `AsString` is is the `cty` equivalent of `.(string)`, and will panic if the value is not a string. I would use [`GoString`](https://pkg.go.dev/github.com/zclconf/go-cty@v1.10.0/cty#Value.GoString), as its expressly designed for this.

3. Let's use the `Details` field of the `hcl.Diagnostic` to inform the user what values are valid.